pkg/model/openstackmodel/servergroup.go

```diff
@@ -44,6 +45,15 @@ var _ fi.ModelBuilder = &ServerGroupModelBuilder{}
 // See https://specs.openstack.org/openstack/nova-specs/specs/newton/approved/lowercase-metadata-keys.html for details
 var instanceMetadataNotAllowedCharacters = regexp.MustCompile("[^a-zA-Z0-9-_:. ]")
 
+// Constants for truncating Tags
+const MAX_TAG_LENGH_OPENSTACK = 60
+
+var TRUNCATE_OPT = truncate.TruncateStringOptions{
+	MaxLength:     MAX_TAG_LENGH_OPENSTACK,
+	AlwaysAddHash: false,
+	HashLength:    6,
+}
+
```

Comment:
@akkina2107 What I'd probably do is, consolidate the truncating definitions from [here][1] and [there][2] into the `k8s.io/kops/upup/pkg/fi/cloudup/openstack` package. Either to [`cloudup/openstack/port.go`][3] or [`cloudup/openstack/utils.go`][4].

For example there are already some consts for instances handled like this in [`cloudup/openstack/instance.go`][5], which are used "all over the place".

Also some typo fix:
```suggestion
// Constants for truncating Tags
const MAX_PORT_TAG_LENGTH_OPENSTACK = 60

var TRUNCATE_OPT = truncate.TruncateStringOptions{
	MaxLength:     MAX_PORT_TAG_LENGTH_OPENSTACK,
	AlwaysAddHash: false,
	HashLength:    6,
}
```

Anyhow, I'm basically just suggesting, and have no authority to approve anything etc. Just trying to give some (hopefully) constructive feedback, since I ran into this issue too and wanted to fix it myself. In the end it's up to the official maintainers.

And speaking of them, AFAIK some of them are currently enjoying their vacation or are generally on some other sort of leave.

As for bringing this into 1.22. I am not sure if this will be done, but probably @hakman can clarify this. Or you simply can build your own forked version with it (with some rudimentary CI) like we do with some changes which are not backported.

[1]: https://github.com/kubernetes/kops/blob/eb7c9ab5495839455038c251c80140cc68301eb2/pkg/model/openstackmodel/servergroup.go#L48-L55
[2]: https://github.com/kubernetes/kops/blob/eb7c9ab5495839455038c251c80140cc68301eb2/upup/pkg/fi/cloudup/openstacktasks/instance.go#L66-L73
[3]: https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/openstack/port.go
[4]: https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/openstack/utils.go
[5]: https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/openstack/instance.go#L40-L47